### PR TITLE
upstream: describe the upstream/downstream relationship to OpenHarmony

### PR DESCRIPTION
**Pull request by Stefan-Schmidt**
High-lebel describition on how Oniro relates to OpenHarmnony. How it handles forking in the downstream distribution.

### CODE REVIEW

File: upstream.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+downstream fork will hold all changes and will be tested by developers and CI for
+releases.
+
+This fork is no long-term home for most of the changes, though.
+Any change that is also applicable to OpenHarmony upstream we need to propose a
+pull request against the master branch in Gitee. By doing so we get crucial
+feedback from OpenHarmony developers. This will help in the design and
+implementation of changes to fit well into the existing code base. And secondly,
+any change that can be upstreamed will come back to Eclipse Oniro with the next
+release sync and reduce the maintenance burden to carry patches downstream.
```
Comment (frankplus) [inline upstream.md:48]:
```suggestion
any changes viable for upstream inclusion will return to Eclipse Oniro with the next
release synchronization, reducing the maintenance burden to carry patches downstream.
```

File: upstream.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+In case a fix was upstreamed we should also work on backporting it into the
+current release to allow us getting it back into our flow in the current cycle.
```
Comment (frankplus) [inline upstream.md:51]:
```suggestion
Should a fix be integrated upstream, efforts should be directed towards backporting 
it into the current release. This ensures its inclusion in our ongoing cycle.
```

**Review by frankplus — CHANGES_REQUESTED** [review]
I am leaving some change suggestions mainly to improve readability.

**Comment by Stefan-Schmidt** [comment]
This one was a bit rough. Thanks for all the suggestions. Integrated and pushed again.

**Review by frankplus — APPROVED** [review]
Looks good!

**Pull request merged**